src/main/java/com/st/utopia/cancellationservice/model/Ticket.java

```diff
@@ -0,0 +1,166 @@
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+/**
+ * A Ticket model.
+ *
+ * 
+ * Model used for the purpose of the cancellation microservice.
+ * 
+ * There are no set methods for information about the ticket such as flight number, row/seat, or class.
+ * 
+ * Upon cancelling the ticket, the microservice needs to set the following fields as null:
+ * 	-reserver
+ * 	-price
+ * 	-reservation_timeout
+ * 	-booking_id
+ * 
+ * This model does not need to define a composite key since the cancellation microservice will search for the ticket via booking_id which is unique
+ * 
+ * 
+ * @author Al-amine AHMED MOUSSA
+ */ 
+
+@Entity
+@Table(name="tbl_tickets")
+public class Ticket{
+	/**
+	 * The three fields that together constitute the identity of a Ticket.
+	 */
+	@EmbeddedId
+	private TicketIdentity id;
+	
+	@Column(name="class")
+	private Integer tier; //class is a keyword 
+
+	private Integer reserver;
+		
+	private Integer price;
+	
+	@Column(name="reservation_timeout")
+	private LocalDateTime reservationTimeout;
+	
+	@Column(name="booking_id")
+	private String bookingId;
+	/**
+	 * the Getters and Setters of the Class
+	 */
+	public TicketIdentity getId() {
+		return id;
+	}
+
+	public Integer getFlight() {
+		if (id == null) {
+			return null;
+		} else {
+			return id.getFlight();
+		}
+	}
+	
+	public Integer getRow() {
+		if (id == null) {
+			return null;
+		} else {
+			return id.getRow();
+		}
+	}
+	
+	public char getSeat() {
+		if (id == null) {
+			return 0;
+		} else {
+			return id.getSeat();
+		}
+	}
+
+	public Integer getTier() {
+		return tier;
+	}
+
+	public void setTier(Integer tier) {
+		this.tier = tier;
+	}
+
+	public Integer getReserver() {
+		return reserver;
+	}
+
+	public void setReserver(Integer reserver) {
+		this.reserver = reserver;
+	}
+
+	public Integer getPrice() {
+		return price;
+	}
+
+	public void setPrice(Integer price) {
+		this.price = price;
+	}
+
+	public LocalDateTime getReservationtimeout() {
+		return reservationTimeout;
+	}
+
+	public void setReservationtimeout(LocalDateTime reservationTimeout) {
+		this.reservationTimeout = reservationTimeout;
+	}
+
+	public String getBookingid() {
+		return bookingId;
+	}
+
+	public void setBookingid(String bookingId) {
+		this.bookingId = bookingId;
+	}
+	
+	/*
+	 * we need empty Constructor for Hibernate Jpa Purposes 
+	 */ 
+	public Ticket() {
+	}
+
+	public Ticket(TicketIdentity id) {
+		this.id = id;
+	}
+
+	public Ticket(TicketIdentity id, Integer tier, Integer reserver, Integer price, LocalDateTime reservationTimeout,
+			String bookingId) {
+		this.id = id;
+		this.tier = tier;
+		this.reserver = reserver;
+		this.price = price;
+		this.reservationTimeout = reservationTimeout;
+		this.bookingId = bookingId;
+	}
+	
+	@Override
+	public int hashCode() {
+		return Objects.hash(id);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
```

Comment:
It's more concise and more robust in the face of subclasses (like the ones the JPA layer will generate) to, instead of checking against `null` and then checking equality of the results of the `getClass()` method, to check whether `obj` is an instance of this class, like so:

```java
@Override
public boolean equals(Object obj) {
        if (this == obj) {
                return true;
        } else if (obj instanceof Ticket) {
                return Objects.equals(id, ((Ticket) obj).id);
        }
}
```